packages/sdk/src/rtl/browserTransport.ts

```diff
@@ -46,22 +46,114 @@ import { BaseTransport } from './baseTransport'
 import { lookerVersion } from './constants'
 import { ICryptoHash } from './cryptoHash'
 
-async function parseResponse(res: IRawResponse) {
+/** Does this browser have the necessary performance APIs? */
+const performs =
+  performance &&
+  performance.mark !== undefined &&
+  performance.measure !== undefined
+
+const startMark = 'A'
+const endMark = 'B'
+
+/**
+ * Create a performance mark
+ * @param name to use as prefix of mark. Use `markName()` to determine the name
+ * @param tag tag to use to distinguish mark
+ */
+const mark = (name: string, tag: string) => {
+  if (performs) {
+    performance.mark(`${name}-${tag}`)
+  }
+}
+
+/**
+ * Create a unique name for the performance metric of the url based on its start time
+ * @param url metric to find
+ *
+ * Uses the last entry of the matching metric to create the name for the tracker so
+ * post processing can use it to merge into the main resource tracker
+ *
+ */
+const markName = (url: string) => {
+  if (!performs) return url
+
+  const entries = performance.getEntriesByName(url, 'resource')
+  if (entries.length > 0) {
+    const last = entries[entries.length - 1]
+    return `${url}-${last.startTime}`
+  }
+  return url
+}
+
+/**
+ * Mark the start of a performance measure
+ * @param name to use as prefix of mark. Use `markName()` to determine the name
+ */
+const markStart = (name: string) => {
+  mark(name, startMark)
+}
+
+/**
+ * Remove entry if it exists
+ * @param name to remove
+ * @param type to match
+ */
+const removePerf = (name: string, type: string) => {
+  if (performance.getEntriesByName(name, type).length !== 0) {
+    // prevent duplicate measures
+    performance.clearMarks(name)
+  }
+}
+
+/**
+ * Mark the end of a performance measure
+ * @param url to find starting mark which doesn't have a resource entry yet
+ * @param name to use as prefix of mark. Use `markName()` to determine the name
+ *
+ * Also creates the start/end performance measure
+ */
+const markEnd = (url: string, name: string) => {
+  if (performs) {
+    const end = `${name}-${endMark}`
+    removePerf(end, 'mark')
+    mark(name, endMark)
+    let start = `${name}-${startMark}`
+    if (performance.getEntriesByName(start, 'mark').length === 0) {
+      start = `${url}-${startMark}`
+    }
+    removePerf(name, 'measure')
+    performance.measure(`${name}`, start, end)
+  }
+}
```

Comment:
This code gets mildly ambiguous because sometimes the timing resource for the _start_ is already in existence so the name is more precise, and sometimes it's not. Also, depending on whether the end-user uses `rawRequest()` by itself or followed by `request()` we may need to ignore the previous _end_ mark.